astTraversal/tags.go

```diff
@@ -5,46 +5,88 @@ import (
 	"strings"
 )
 
-func ParseStructTag(tag string) (name string, isRequired bool, isShown bool) {
-	isShown = true
-	if tag != "" {
-		binding := reflect.StructTag(tag).Get("binding")
-		if binding != "" {
-			isRequired = strings.Contains(binding, "required")
+type BindingTagType string
+
+const (
+	HeaderBindingTag BindingTagType = "header"
+	FormBindingTag   BindingTagType = "form"
+	URIBindingTag    BindingTagType = "uri"
+	JSONBindingTag   BindingTagType = "json"
+	XMLBindingTag    BindingTagType = "xml"
+	YAMLBindingTag   BindingTagType = "yaml"
+	NoBindingTag     BindingTagType = ""
+)
+
+var BindingTags = []BindingTagType{HeaderBindingTag, FormBindingTag, URIBindingTag, JSONBindingTag, XMLBindingTag, YAMLBindingTag}
+
+type BindingTag struct {
+	Name       string
+	IsShown    bool
+	IsOptional bool
+}
+
+type BindingTagMap map[BindingTagType]BindingTag
+
+type ValidationTagType string
+
+const (
+	GinValidationTag       ValidationTagType = "binding"
+	ValidatorValidationTag ValidationTagType = "validate"
+	NoValidationTag        ValidationTagType = ""
```

Comment:
Can we add a format tag here

Reply:
What would a format tag give us here, eventually all of that type of information would be contained in the gin and validator validation tags

Reply:
True, I'm thinking manual overrides. But if you don't think it's required, that's fair

Reply:
Yeah that's fair but I think we probably need a bit more thought before we add something like that, hopefully we shouldn't need it aha and also I think it would work slightly differently as it would probably fall under a different tag category. If you look at them moment all of the tag categories all have the same format and spec so the logic can be used across them.

Reply:
Ok sounds good